openqaoa/problems/problem.py

```diff
@@ -1144,34 +1037,24 @@ def __init__(self, G, source, dest):
         assert source != dest, "Source and destination nodes cannot be the same"
 
     @staticmethod
-    def random_instance(**kwargs):
+    def random_instance(problem_size):
         """
         Creates a random instance of the Shortest problem, whose graph is
-        random following the Erdos-Renyi model. By default the node and edge 
+        random following the Erdos-Renyi model. By default the node and edge
         weights are set to 1.0 and the default constraint is taken to be as large.
 
-        Parameters
-        ----------
-        **kwargs:
-            Required keyword arguments are:
-
-            n_nodes: int
-                The number of nodes (vertices) in the graph.
-            edge_probability: float
-                The probability with which an edge is added to the graph.
 
         Returns
         -------
         A random instance of the Shortest Path problem.
         """
-
-        n_nodes, edge_probability, seed, source, dest = check_kwargs(
-            ["n_nodes", "edge_probability", "seed", "source", "dest"],
-            [None, None, 1234, 0, 1],
-            **kwargs,
-        )
+        np.random.seed(SEED)
+        n_nodes = np.random.randint(1, problem_size)
+        edge_probability = np.random.uniform(0, 1)
+        source = np.random.randint(0, n_nodes)
+        dest = np.random.randint(0, n_nodes)
```

Comment:
If this function generates a source and dest node of the same index, it will error out since the ShortestPath class requires the nodes to have a different index.